### ISSUE TITLE
Missing closing marimo-filename tag when exporting with MarimoIslandGenerator

### ISSUE THREAD

**Issue by mattiasJohnson** [marimo-team/marimo#2397]
### Describe the bug

When running pretty much the example code from https://docs.marimo.io/guides/exporting.html 

```py
from marimo import MarimoIslandGenerator

async def build_app():
    generator = MarimoIslandGenerator()
    block1 = generator.add_code("import marimo as mo")
    block2 = generator.add_code("mo.md('Hello, islands!')")

    app = await generator.build()

    output = f"""
    <html>
        <head>
            {generator.render_head()}
        </head>
        <body>
            {block1.render(display_output=False)}
            {block2.render()}
        </body>
    </html>
    """
    return output

import asyncio
output = asyncio.run(build_app())
with open("marimo.html", "w") as file:
    file.write(output)
```

I do not get the closing tag of marimo-filename:

```html

    <html>
        <head>
            <script type="module" src="https://cdn.jsdelivr.net/npm/@marimo-team/islands@0.8.15/dist/main.js"></script>
    <link
        href="https://cdn.jsdelivr.net/npm/@marimo-team/islands@0.8.15/dist/style.css"
        rel="stylesheet"
        crossorigin="anonymous"
    />
    <link rel="preconnect" href="https://fonts.googleapis.com" />
    <link
        rel="preconnect"
        href="https://fonts.gstatic.com"
        crossorigin
    />
    <link href="https://fonts.googleapis.com/css2?family=Fira+Mono:wght@400;500;700&amp;family=Lora&amp;family=PT+Sans:wght@400;700&amp;display=swap" rel="stylesheet" />
    <link
        rel="stylesheet"
        href="https://cdn.jsdelivr.net/npm/katex@0.16.10/dist/katex.min.css"
        integrity="sha384-wcIxkf4k558AjM3Yz3BBFQUbk/zgIYC2R0QpeeYb+TwlBVMrlgLqwRjRtGZiK7ww"
        crossorigin="anonymous"
    />
    <marimo-filename hidden></>
<marimo-mode data-mode='read' hidden></marimo-mode>
        </head>
        <body>
            <marimo-island
    data-app-id="main"
    data-cell-id="Hbol"
    data-reactive="true"
>
    <marimo-cell-output>
    </marimo-cell-output>
    <marimo-cell-code hidden>import%20marimo%20as%20mo</marimo-cell-code>
</marimo-island>
            <marimo-island
    data-app-id="main"
    data-cell-id="MJUe"
    data-reactive="true"
>
    <marimo-cell-output>
    <span class="markdown prose dark:prose-invert"><span class="paragraph">Hello, islands!</span></span>
    </marimo-cell-output>
    <marimo-cell-code hidden>mo.md('Hello%2C%20islands!')</marimo-cell-code>
</marimo-island>
        </body>
    </html>
    
```

### Environment

{
  "marimo": "0.8.15",
  "OS": "Linux",
  "OS Version": "6.8.0-45-generic",
  "Processor": "x86_64",
  "Python Version": "3.12.5",
  "Binaries": {
    "Browser": "--",
    "Node": "--"
  },
  "Dependencies": {
    "click": "8.1.7",
    "importlib-resources": "missing",
    "jedi": "0.19.1",
    "markdown": "3.7",
    "pygments": "2.18.0",
    "pymdown-extensions": "10.9",
    "ruff": "0.6.4",
    "starlette": "0.38.4",
    "tomlkit": "0.13.2",
    "typing-extensions": "4.12.2",
    "uvicorn": "0.30.6",
    "websockets": "12.0"
  },
  "Optional Dependencies": {
    "altair": "5.4.1"
  }
}

### Code to reproduce

Steps in description above.

**Comment by mattiasJohnson** [comment]
Using `MarimoIslandGenerator.from_file` does not give a closing `marimo-filename` either.

**Comment by mscolnick** [comment]
This should be fixed already on the latest version. Can you upgrade your version?

**Comment by mattiasJohnson** [comment]
Excuse me that was a miss by me, confirming that it is indeed fixed in marimo 0.8.19. Thanks for a great project!
